src/value/borrowed.rs

```diff
@@ -91,221 +80,251 @@ impl<'v> Value<'v> {
+            Self::Static(StaticNode::Null) => true,
             _ => false,
         }
     }
 
+    #[inline]
     fn as_bool(&self) -> Option<bool> {
         match self {
-            Value::Bool(b) => Some(*b),
+            Self::Static(StaticNode::Bool(b)) => Some(*b),
             _ => None,
         }
     }
 
+    #[inline]
     fn as_i64(&self) -> Option<i64> {
         match self {
-            Value::I64(i) => Some(*i),
+            Self::Static(StaticNode::I64(i)) => Some(*i),
+            Self::Static(StaticNode::U64(i)) => i64::try_from(*i).ok(),
             _ => None,
         }
     }
 
+    #[inline]
     fn as_u64(&self) -> Option<u64> {
         #[allow(clippy::cast_sign_loss)]
         match self {
-            Value::I64(i) if *i >= 0 => Some(*i as u64),
+            Self::Static(StaticNode::I64(i)) => u64::try_from(*i).ok(),
+            Self::Static(StaticNode::U64(i)) => Some(*i),
             _ => None,
         }
     }
 
+    #[inline]
     fn as_f64(&self) -> Option<f64> {
         match self {
-            Value::F64(i) => Some(*i),
+            Self::Static(StaticNode::F64(i)) => Some(*i),
             _ => None,
         }
     }
 
+    #[inline]
     fn cast_f64(&self) -> Option<f64> {
         #[allow(clippy::cast_precision_loss)]
         match self {
-            Value::F64(i) => Some(*i),
-            Value::I64(i) => Some(*i as f64),
-            _ => None,
-        }
-    }
-
-    fn as_string(&self) -> Option<String> {
-        match self {
-            Value::String(s) => Some(s.to_string()),
+            Self::Static(StaticNode::F64(i)) => Some(*i),
+            Self::Static(StaticNode::I64(i)) => Some(*i as f64),
+            Self::Static(StaticNode::U64(i)) => Some(*i as f64),
             _ => None,
         }
     }
 
+    #[inline]
     fn as_str(&self) -> Option<&str> {
         use std::borrow::Borrow;
         match self {
-            Value::String(s) => Some(s.borrow()),
+            Self::String(s) => Some(s.borrow()),
             _ => None,
         }
     }
 
+    #[inline]
     fn as_array(&self) -> Option<&Vec<Value<'v>>> {
         match self {
-            Value::Array(a) => Some(a),
+            Self::Array(a) => Some(a),
             _ => None,
         }
     }
 
+    #[inline]
     fn as_array_mut(&mut self) -> Option<&mut Vec<Value<'v>>> {
         match self {
-            Value::Array(a) => Some(a),
+            Self::Array(a) => Some(a),
             _ => None,
         }
     }
 
+    #[inline]
     fn as_object(&self) -> Option<&HashMap<Self::Key, Self>> {
         match self {
-            Value::Object(m) => Some(m),
+            Self::Object(m) => Some(m),
             _ => None,
         }
     }
 
+    #[inline]
     fn as_object_mut(&mut self) -> Option<&mut HashMap<Self::Key, Self>> {
         match self {
-            Value::Object(m) => Some(m),
+            Self::Object(m) => Some(m),
             _ => None,
         }
     }
 }
 
+#[cfg_attr(tarpaulin, skip)]
```

Comment:
interesting, why `skip`?

Reply:
It's just a Display implementation that dispatches into the underlying rust types, so if that would go wrong we've waaayyyy bigger problems - aka not worth putting testing effort into it.